### PR TITLE
Initial commit

### PR DESCRIPTION
**Pull request by bpankowski**
Still missing nginx config.

### CODE REVIEW

File: ansible_config.yml [hunk 1]
```diff
@@ -0,0 +1,88 @@
+-
```
Comment (jluk-codi) [inline ansible_config.yml]: Rename this file to "ansible_playbook", ansible config file is something different: https://docs.ansible.com/ansible/latest/installation_guide/intro_configuration.html#configuration-file

**Review by jluk-codi — CHANGES_REQUESTED** [review]
Remove the unnecessary local files:
.vagrant directory
ansible .retry files
virtual machine .log files

Use gitignore files to prevent adding them again: https://git-scm.com/docs/gitignore

File: .gitignore [hunk 2]
```diff
@@ -0,0 +1,3 @@
+.vagrant
+ansible_config.retry
```
Comment (jluk-codi) [inline .gitignore]: This matches retry file only for an ansible playbook named "ansible_config". Change this to "*.retry".

**Review by jluk-codi — CHANGES_REQUESTED** [review]

File: ansible_playbook.yml [hunk 3]
```diff
@@ -0,0 +1,90 @@
+-
+  hosts: all
+  remote_user: vagrant
+  sudo: yes
+  tasks: 
+    - name: "update"
+      apt:
+        update_cache: yes
+    - name: "Install linux-image-extra-"
```
Comment (wurbanski) [inline ansible_playbook.yml]: I think all those tasks used for installing packages can be replaced with one using `with_items:` clause

File: ansible_playbook.yml [hunk 4]
```diff
@@ -0,0 +1,90 @@
...
+      apt:
+        name: software-properties-common
+        state: latest
+    - name: 'Add Docker apt key'
+      apt_key:
+        url: 'https://download.docker.com/linux/ubuntu/gpg'
+        id: 9DC858229FC7DD38854AE2D88D81803C0EBFCD88
+        state: present
+      register: add_repository_key
+      ignore_errors: true
```
Comment (wurbanski) [inline ansible_playbook.yml]: Why do we ignore errors here?

File: Vagrantfile [hunk 5]
```diff
@@ -0,0 +1,75 @@
+# -*- mode: ruby -*-
```
Comment (wurbanski) [inline Vagrantfile]: I think most of these comments are not needed :)

File: ansible_playbook.yml [hunk 6]
```diff
@@ -0,0 +1,90 @@
...
+        pkg: docker-ce
+        state: latest
+        update_cache: yes
+    - name: 'Ensure Docker service is installed and restarted'
+      service:
+        name: docker
+        state: restarted
+        enabled: yes
+    - name: 'Create Dir'
+      file: 'path=/src/www state=directory'
```
Comment (wurbanski) [inline ansible_playbook.yml]: I think you should keep using one notation, and the other (dict) is usually preferred :)

**Review by wurbanski — CHANGES_REQUESTED** [review]

File: ansible_playbook.yml (hunk 6 above)
Comment (diabelko) [reply ansible_playbook.yml]: ++

File: ansible_playbook.yml (hunk 4 above)
Comment (diabelko) [reply ansible_playbook.yml]: I think there's no need to do 'dry-run' for adding apt key. I would suggest skipping this step, ensure curl is present unconditionally and then download the key - if curl is installed nothing will happen and we want to fail anyway if the key won't get downloaded properly.

File: ansible_playbook.yml [hunk 7]
```diff
@@ -0,0 +1,90 @@
...
+      when: add_repository_key|failed
+    - name: 'Add docker apt repo'
+      apt_repository:
+        repo: 'deb [arch=amd64] https://download.docker.com/linux/ubuntu {{ansible_lsb.codename}} stable'
+        update_cache: yes
+    - name: 'Install Docker apt package'
+      apt:
+        pkg: docker-ce
+        state: latest
+        update_cache: yes
```
Comment (diabelko) [inline ansible_playbook.yml]: Updating cache is needed only when making changes to repos or there are newer packages. There's no need to do it after installing packages.

File: ansible_playbook.yml [hunk 8]
```diff
@@ -0,0 +1,90 @@
...
+    - name: 'Ensure Docker service is installed and restarted'
+      service:
+        name: docker
+        state: restarted
```
Comment (diabelko) [inline ansible_playbook.yml]: Why do we need docker `restarted` instead of `started`? Also, will 2nd Ansible run restart Docker again?

**Review by diabelko — CHANGES_REQUESTED** [review]

File: ansible_playbook.yml [hunk 9]
```diff
@@ -0,0 +1,66 @@
...
+- hosts: all
+  remote_user: vagrant
+  sudo: yes
+  tasks: 
+
+    - name: 'update'
+      apt:
+        update_cache: yes
+
+    - name: 'Looping installation of apt'
```
Comment (wurbanski) [inline ansible_playbook.yml]: Technically, It's not an `installation of apt` :P

File: ansible_playbook.yml [hunk 10]
```diff
@@ -0,0 +1,66 @@
...
+        name: '{{ item }}'
+        state: latest
+      with_items:
+        - linux-image-extra-4.13.0-36-generic
+        - linux-image-extra-virtual
+        - apt-transport-https
+        - ca-certificates
+        - software-properties-common
+        - curl
+        - docker-ce
```
Comment (jluk-codi) [inline ansible_playbook.yml]: Ups.

```TASK [Looping installation of apt] *********************************************
failed: [default] (item=[u'linux-image-extra-4.13.0-36-generic', u'linux-image-extra-virtual', u'apt-transport-https', u'ca-certificates', u'software-properties-common', u'curl', u'docker-ce', u'python3', u'python-pip', u'docker']) => {"changed": false, "item": ["linux-image-extra-4.13.0-36-generic", "linux-image-extra-virtual", "apt-transport-https", "ca-certificates", "software-properties-common", "curl", "docker-ce", "python3", "python-pip", "docker"], "msg": "No package matching 'docker-ce' is available"}
	to retry, use: --limit @/vagrant/ansible_playbook.retry

PLAY RECAP *********************************************************************
default                    : ok=2    changed=1    unreachable=0    failed=1```

File: ansible_playbook.yml [hunk 11]
```diff
@@ -0,0 +1,66 @@
...
+        update_cache: yes
+
+    - name: 'Ensure Docker service is installed and started'
+      service:
+        name: 'docker'
+        state: restarted
+        enabled: yes
+
+    - name: 'Create directory'
+      file: 'path=/src/www state=directory'
```
Comment (jluk-codi) [inline ansible_playbook.yml]: Change this to regular YAML dict syntax

**Review by jluk-codi — CHANGES_REQUESTED** [review]

**Comment by jluk-codi** [comment]
Change the author of all the commits to use your main email address. This is an exercise before starting your work with Gerrit, as Gerrit doesn't allow pushing commits with mismatched author. See here for hints: https://stackoverflow.com/a/3042512

File: ansible_playbook.yml (hunk 9 above)
Comment (bpankowski) [reply ansible_playbook.yml]: ++

File: ansible_playbook.yml [hunk 12]
```diff
@@ -0,0 +1,66 @@
...
+        - linux-image-extra-4.13.0-36-generic
+        - linux-image-extra-virtual
+        - apt-transport-https
+        - ca-certificates
+        - software-properties-common
+        - curl
+        - docker-ce
+        - python3
+        - python-pip
+        - docker
```
Comment (jluk-codi) [inline ansible_playbook.yml]: Is this needed while you also install `docker-ce`?

**Review by jluk-codi — CHANGES_REQUESTED** [review]

**Pull request closed without merging**